lib/RemoteAST/InProcessMemoryReader.cpp

```diff
@@ -0,0 +1,41 @@
+//===--- InProcessMemoryReader.cpp - Reads local memory ---------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+//  This file implements the abstract interface for working with remote memory.
+//  This method cannot be implemented in the header, as we must avoid importing
+//  <windows.h> in a header, which causes conflicts with Swift definitions.
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Remote/InProcessMemoryReader.h"
+
+#if defined(_WIN32)
+#define WIN32_LEAN_AND_MEAN
+#define NOMINMAX
+#include <windows.h>
+#else
+#include <dlfcn.h>
+#endif
+
+using namespace swift;
+using namespace swift::remote;
+
+RemoteAddress InProcessMemoryReader::getSymbolAddress(const std::string &name) {
+#if defined(_WIN32)
+    HMODULE module = GetModuleHandle(NULL);
+    auto pointer = GetProcAddress(module, name.c_str());
+    FreeLibrary(module);
```

Comment:
`GetModuleHandle` does **not** increment reference count for the module.  When you are invoking `FreeLibrary` here, you are decrementing the refcount on the module.  This can cause the main module to hit a reference count of 0, and be unloaded!  Am I missing something?